# Лабораторная 1

## 1. Текстовое описание выбранного Вами набора данных.

Был выбран датасет Housing Price Prediction, содеражщий данные о цене дома и его праметрах(метраж, количество ванных, наличие парковочных мест и т.д.)

In [ ]:
%matplotlib inline
%config InlineBackend.figure_format='retina'

import json

import matplotlib.pyplot as plt

import numpy as np
import pandas as pd
pd.set_option('display.max_columns', None)

import ast
import plotly.graph_objs as go
from plotly.subplots import make_subplots
import plotly.express as px
import seaborn as sns

In [ ]:
data = pd.read_csv("/Housing.csv")
data.head(5)

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  
0              no             yes        2      yes        furnished  
1              no             yes        3       no        furnished  
2              no              no        2      yes   semi-furnished  
3              no             yes        3      yes        furnished  
4              no             yes        2       no        furnished

## 2. Основные характеристики датасета.

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB


## 3. Визуальное исследование датасета.

In [ ]:
fig = make_subplots(rows=5, cols=2, subplot_titles=('Bathrooms vs Price',
                                                    'Air Conditioning vs Price',
                                                    'Hot Water Heating vs Price',
                                                    'Preferred Area vs Price',
                                                    'Stories vs Price',
                                                    'Main Road vs Price',
                                                    'Basement vs Price',
                                                    'Guestroom vs Price',
                                                    'Parking vs Price',
                                                    'Furnishing Status vs Price'),
                    vertical_spacing=0.08,
                    horizontal_spacing=0.1)

box_plot_bathrooms = go.Box(x=data['bathrooms'], y=data['price'],
                            marker=dict(color='blue'), name='Bathrooms vs Price')

box_plot_air_conditioning = go.Box(x=data['airconditioning'], y=data['price'],
                                   marker=dict(color='orange'), name='Air Conditioning vs Price')

box_plot_hot_water_heating = go.Box(x=data['hotwaterheating'], y=data['price'],
                                    marker=dict(color='green'), name='Hot Water Heating vs Price')

box_plot_pref_area = go.Box(x=data['prefarea'], y=data['price'],
                            marker=dict(color='red'), name='Preferred Area vs Price')

box_plot_stories = go.Box(x=data['stories'], y=data['price'],
                          marker=dict(color='purple'), name='Stories vs Price')

box_plot_mainroad = go.Box(x=data['mainroad'], y=data['price'],
                           marker=dict(color='cyan'), name='Main Road vs Price')

box_plot_basement = go.Box(x=data['basement'], y=data['price'],
                           marker=dict(color='gray'), name='Basement vs Price')

box_plot_guestroom = go.Box(x=data['guestroom'], y=data['price'],
                            marker=dict(color='pink'), name='Guestroom vs Price')

box_plot_parking = go.Box(x=data['parking'], y=data['price'],
                          marker=dict(color='brown'), name='Parking vs Price')

box_plot_furnishing_status = go.Box(x=data['furnishingstatus'], y=data['price'],
                                    marker=dict(color='yellow'), name='Furnishing Status vs Price')

fig.add_trace(box_plot_bathrooms, row=1, col=1)
fig.add_trace(box_plot_air_conditioning, row=1, col=2)
fig.add_trace(box_plot_hot_water_heating, row=2, col=1)
fig.add_trace(box_plot_pref_area, row=2, col=2)
fig.add_trace(box_plot_stories, row=3, col=1)
fig.add_trace(box_plot_mainroad, row=3, col=2)
fig.add_trace(box_plot_basement, row=4, col=1)
fig.add_trace(box_plot_guestroom, row=4, col=2)
fig.add_trace(box_plot_parking, row=5, col=1)
fig.add_trace(box_plot_furnishing_status, row=5, col=2)

fig.update_layout(height=2000, width=1200,
                  title_text="Comprehensive Visualization of House Prices",
                  title_x=0.5,
                  showlegend=False)


fig.update_xaxes(title_text="Number of Bathrooms", row=1, col=1)
fig.update_xaxes(title_text="Air Conditioning", row=1, col=2)
fig.update_xaxes(title_text="Hot Water Heating", row=2, col=1)
fig.update_xaxes(title_text="Preferred Area", row=2, col=2)
fig.update_xaxes(title_text="Number of Stories", row=3, col=1)
fig.update_xaxes(title_text="Proximity to Main Road", row=3, col=2)
fig.update_xaxes(title_text="Basement", row=4, col=1)
fig.update_xaxes(title_text="Guestroom", row=4, col=2)
fig.update_xaxes(title_text="Parking", row=5, col=1)
fig.update_xaxes(title_text="Furnishing Status", row=5, col=2)

fig.update_yaxes(title_text="Price", col=1)

fig.update_annotations(font=dict(size=20))

fig.show()

In [ ]:
min_price_bathrooms = data.groupby('bathrooms')['price'].mean().min()
avg_price_bathrooms = data.groupby('bathrooms')['price'].mean()
avg_increase_bathrooms = ((avg_price_bathrooms / min_price_bathrooms) - 1) * 100
bathrooms_increase_dict = avg_increase_bathrooms.to_dict()
print("Average Increase Percentage for Number of Bathrooms:")
for key, value in sorted(bathrooms_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_air_conditioning = data.groupby('airconditioning')['price'].mean().min()
avg_price_air_conditioning = data.groupby('airconditioning')['price'].mean()
avg_increase_air_conditioning = ((avg_price_air_conditioning / min_price_air_conditioning) - 1) * 100
air_conditioning_increase_dict = avg_increase_air_conditioning.to_dict()
print("\nAverage Increase Percentage for Air Conditioning:")
for key, value in sorted(air_conditioning_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_hot_water_heating = data.groupby('hotwaterheating')['price'].mean().min()
avg_price_hot_water_heating = data.groupby('hotwaterheating')['price'].mean()
avg_increase_hot_water_heating = ((avg_price_hot_water_heating / min_price_hot_water_heating) - 1) * 100
hot_water_heating_increase_dict = avg_increase_hot_water_heating.to_dict()
print("\nAverage Increase Percentage for Hot Water Heating:")
for key, value in sorted(hot_water_heating_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_pref_area = data.groupby('prefarea')['price'].mean().min()
avg_price_pref_area = data.groupby('prefarea')['price'].mean()
avg_increase_pref_area = ((avg_price_pref_area / min_price_pref_area) - 1) * 100
pref_area_increase_dict = avg_increase_pref_area.to_dict()
print("\nAverage Increase Percentage for Preferred Area:")
for key, value in sorted(pref_area_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_stories = data.groupby('stories')['price'].mean().min()
avg_price_stories = data.groupby('stories')['price'].mean()
avg_increase_stories = ((avg_price_stories / min_price_stories) - 1) * 100
stories_increase_dict = avg_increase_stories.to_dict()
print("\nAverage Increase Percentage for Stories:")
for key, value in sorted(stories_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_mainroad = data.groupby('mainroad')['price'].mean().min()
avg_price_mainroad = data.groupby('mainroad')['price'].mean()
avg_increase_mainroad = ((avg_price_mainroad / min_price_mainroad) - 1) * 100
mainroad_increase_dict = avg_increase_mainroad.to_dict()
print("\nAverage Increase Percentage for Main Road:")
for key, value in sorted(mainroad_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_basement = data.groupby('basement')['price'].mean().min()
avg_price_basement = data.groupby('basement')['price'].mean()
avg_increase_basement = ((avg_price_basement / min_price_basement) - 1) * 100
basement_increase_dict = avg_increase_basement.to_dict()
print("\nAverage Increase Percentage for Basement:")
for key, value in sorted(basement_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_guestroom = data.groupby('guestroom')['price'].mean().min()
avg_price_guestroom = data.groupby('guestroom')['price'].mean()
avg_increase_guestroom = ((avg_price_guestroom / min_price_guestroom) - 1) * 100
guestroom_increase_dict = avg_increase_guestroom.to_dict()
print("\nAverage Increase Percentage for Guestroom:")
for key, value in sorted(guestroom_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_parking = data.groupby('parking')['price'].mean().min()
avg_price_parking = data.groupby('parking')['price'].mean()
avg_increase_parking = ((avg_price_parking / min_price_parking) - 1) * 100
parking_increase_dict = avg_increase_parking.to_dict()
print("\nAverage Increase Percentage for Parking:")
for key, value in sorted(parking_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

min_price_furnishing_status = data.groupby('furnishingstatus')['price'].mean().min()
avg_price_furnishing_status = data.groupby('furnishingstatus')['price'].mean()
avg_increase_furnishing_status = ((avg_price_furnishing_status / min_price_furnishing_status) - 1) * 100
furnishing_status_increase_dict = avg_increase_furnishing_status.to_dict()
print("\nAverage Increase Percentage for Furnishing Status:")
for key, value in sorted(furnishing_status_increase_dict.items()):
    print(f"{key}: {value:.0f}%")

Average Increase Percentage for Number of Bathrooms:
1: 0%
2: 48%
3: 73%
4: 191%

Average Increase Percentage for Air Conditioning:
no: 0%
yes: 43%

Average Increase Percentage for Hot Water Heating:
no: 0%
yes: 18%

Average Increase Percentage for Preferred Area:
no: 0%
yes: 33%

Average Increase Percentage for Stories:
1: 0%
2: 14%
3: 36%
4: 73%

Average Increase Percentage for Main Road:
no: 0%
yes: 47%

Average Increase Percentage for Basement:
no: 0%
yes: 16%

Average Increase Percentage for Guestroom:
no: 0%
yes: 27%

Average Increase Percentage for Parking:
0: 0%
1: 25%
2: 43%
3: 42%

Average Increase Percentage for Furnishing Status:
furnished: 37%
semi-furnished: 22%
unfurnished: 0%


## 4. Информация о корреляции признаков.

 - Количество ванных комнат:
Дома с большим количеством ванных комнат значительно подорожают.
Стоимость домов с 4 ванными комнатами значительно возрастает - на 191% по сравнению с домами с 1 ванной комнатой.
 - Кондиционер:
Средняя цена на недвижимость с системами кондиционирования воздуха заметно выросла - на 43%.
 - Отопление горячей водой:
Средняя стоимость домов с водяным отоплением выросла на 18%.
 - Предпочтительный район:
Средняя цена на недвижимость в предпочтительных районах значительно выросла - на 33%.
 - Этажность:
4-этажный дом имеет существенное увеличение на 73% по сравнению с одноэтажным домом.
 - Близость к главной дороге:
Средняя цена на дома, расположенные вдоль главных дорог, выросла на 47%.
 - Цокольный этаж и комната для гостей:
Стоимость цокольного этажа увеличивается на 16%, в то время как стоимость комнаты для гостей увеличивается в среднем на 27%.
 - Парковка:
Количество парковочных мест в домах значительно увеличилось, при этом количество объектов с 3 парковочными местами увеличилось на 42% по сравнению с отсутствием таковых.
 - Статус меблировки:
Больше всего подорожала недвижимость с отделкой, средняя цена на которую выросла на 37%. За ней следует недвижимость с отделкой, стоимость которой выросла на 22% по сравнению с недвижимостью без отделки